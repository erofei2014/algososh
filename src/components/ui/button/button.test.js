import React from 'react';
import { Button } from './button';
import renderer from 'react-test-renderer';

describe("Компонент Button", () => {
  it("Отрисовка кнопки с текстом", () => {
    const tree = renderer
    .create(<Button text="Текст" />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка кнопки без текста", () => {
    const tree = renderer
    .create(<Button />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка заблокированной кнопки", () => {
    const tree = renderer
    .create(<Button disabled={true} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Отрисовка кнопки с лоадером", () => {
    const tree = renderer
    .create(<Button isLoader={true} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Вызов колбека при клике на кнопку", () => {
    const onClickMock = jest.fn();
    const tree = renderer
    .create(<Button text="Текст" onClick={onClickMock} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});