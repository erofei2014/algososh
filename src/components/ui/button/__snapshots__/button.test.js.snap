// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Button Вызов колбека при клике на кнопку 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  onClick={[MockFunction]}
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;

exports[`Компонент Button Отрисовка заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Компонент Button Отрисовка кнопки без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Компонент Button Отрисовка кнопки с лоадером 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Компонент Button Отрисовка кнопки с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст
  </p>
</button>
`;
